heartbeat/monitors/active/http/check.go

```diff
@@ -84,18 +84,17 @@ func checkHeaders(headers map[string]string) RespCheck {
 	}
 }
 
-func checkBody(body []byte) RespCheck {
+func checkBody(body []match.Matcher) RespCheck {
 	return func(r *http.Response) error {
-		// read up to len(body)+1 bytes for comparing content to be equal
-		in := io.LimitReader(r.Body, int64(len(body))+1)
-		content, err := ioutil.ReadAll(in)
+		content, err := ioutil.ReadAll(r.Body)
 		if err != nil {
 			return err
 		}
-
-		if !bytes.Equal(body, content) {
-			return errBodyMismatch
+		for _, m := range body {
+			if m.MatchString(string(content)) {
```

Comment:
use `m.Match(content)`, so to reduce the chance of allocating and copying all contents, just for matching.

Reply:
Fixed.